cmd/restic/cmd_check.go

```diff
@@ -479,3 +506,15 @@ func selectRandomPacksByFileSize(allPacks map[restic.ID]int64, subsetSize int64,
 	packs := selectRandomPacksByPercentage(allPacks, subsetPercentage)
 	return packs
 }
+
+type checkResult struct {
+	Error           bool     `json:"error"`
+	NumIndexErrors  int      `json:"num_index_errors"`
+	NumPackErrors   int      `json:"num_pack_errors"`
+	NumTreeErrors   int      `json:"num_tree_errors"`
+	NumUnusedBlobs  int      `json:"num_unused_blobs"`
+	NumDataErrors   int      `json:"num_data_errors"`
```

Comment:
Should we merge these?
```suggestion
	NumErrors       int     `json:"num_errors"`
```

Reply:
There's at least the category of snapshot errors missing. The main question here is what one can do with  knowing that there are 5 data errors in contrast to 5 index errors. In regards to index errors one could run `repair index` but giving you that information is the task of `suggest_repair_index`. So, I'm leaning somewhat towards only keeping the overall number of errors.

Reply:
I merged this into a single error count.